include/nlohmann/detail/meta/type_traits.hpp

```diff
@@ -454,6 +468,51 @@ struct is_constructible_tuple : std::false_type {};
 template<typename T1, typename... Args>
 struct is_constructible_tuple<T1, std::tuple<Args...>> : conjunction<is_constructible<T1, Args>...> {};
 
+template<typename BasicJsonType, typename T>
+struct is_json_iterator : std::false_type {};
```

Comment:
Should this be called `is_json_iterator_of` to better reflect that the trait takes both a `BasicJsonType` and an iterator?

Reply:
Good idea. I'm not sure about the argument order then. `is_json_iterator_of<json, it>` or `is_json_iterator_of<it, json>`?

Reply:
"of" referrers to the `BasicJsonType` so it makes more sense as first argument to me.